grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/JwtParser.java

```diff
@@ -0,0 +1,77 @@
+package org.hypertrace.core.grpcutils.context;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class JwtParser {
+  private static final Logger LOG = LoggerFactory.getLogger(JwtParser.class);
+  private static final String BEARER_TOKEN_PREFIX = "Bearer ";
+
+  private final Cache<String, Optional<Jwt>> jwtCache =
```

Comment:
is the cache being still used ?

Reply:
Yes, using the cache to help avoid decoding the same token again.